nixos/modules/tasks/network-interfaces.nix

```diff
@@ -1408,9 +1408,11 @@ in
 
     systemd.services.domainname = lib.mkIf (cfg.domain != null) {
       wantedBy = [ "sysinit.target" ];
-      before = [ "sysinit.target" ];
+      before = [ "sysinit.target" "shutdown.target" ];
+      conflicts = [ "shutdown.target" ];
       unitConfig.DefaultDependencies = false;
       serviceConfig.ExecStart = ''${pkgs.nettools}/bin/domainname "${cfg.domain}"'';
+      serviceConfig.Type = "oneshot";
```

Comment:
While we're at it, we should probably also add RemainAfterExit

Reply:
Only if it makes sense. I don't really know anything about this service. Should it be considered present from the completion of this service? Like, is there a reason to consider it running, or is there a stop procedure should when the system shuts down?

Reply:
[domainname](https://net-tools.sourceforge.io/man/hostname.1.html) sets the NIS domain name. It's fundamentally a call to [setdomainname(2)](https://linux.die.net/man/2/setdomainname).

I don't think it should [RemainAfterExit=yes](https://www.freedesktop.org/software/systemd/man/latest/systemd.service.html#RemainAfterExit=) since it really is just a one-shot configuration.

Reply:
I think the question is, does it make sense to start it multiple times during a boot cycle? If it doesn't, then it should RemainAfterExit imo. Since in my understanding domainname is idempotent, it should RemainAfterExit.

Reply:
> does it make sense to start it multiple times during a boot cycle? If it doesn't, then it should RemainAfterExit imo.

What? No, that's not what that's for. `RemainAfterExit` is there so that things like `ExecStop` can have meaning for oneshot logic (and also so that dependent units can have `BindsTo` mean something). If we need a way to designate oneshots that shouldn't be started again by `switch-to-configuration`, we should make another custom `X-` thing like `X-NoStartOnSwitch` or something.